### DISCUSSION TITLE
PackageIdentifier of OSS packages

### DISCUSSION THREAD

**Discussion by vedantmgoyal9** [microsoft/winget-pkgs#29973] [Q&A] [answered]
I have recently observed that open-source packages do not have a fixed convention for PackageIdentifier. They follow mixed conventions:

- `<repo-owner-id>.<app-name>` (eg. JanDeDobbeleer.OhMyPosh) (repo: JanDeDobbeleer/oh-my-posh)
- `<repo-owner-name>.<app-name>` (I couldn't remember any while starting this discussion but I am sure there were some packages which followed this convention)
- `<publisher-in-arp>.<app-name>` (eg. PaulSori.mStreamServer) (repo: IrosTheBeggar/mStream)
- `<app-name>.<app-name>` (eg. Joplin.Joplin) (repo: laurent22/joplin)

IMHO, there should be a certain uniformity for determining PackageIdentifier for OSS packages. I personally like the first one because it's simple and easy to understand. This would not only help in keeping the repository organised but also in creating a better user experience.

**Comment by KaranKad** [comment, marked as answer]
`<publisher-in-arp>.<app-name>` is what I would vote for without Corp, Inc, LLC etc for `<publisher-in-arp>`.

**Comment by jedieaston** [comment]
Either @KaranKad's suggestion, or `<publisher in signing certificate>.<app-name>`
